app/models/platform_table_info.rb

```diff
@@ -0,0 +1,119 @@
+
+  def initialize(table_stats)
+    @name = table_stats['name']
+    @rows = table_stats['count'].to_i
+    @rows_trend = table_stats['count_trend']
+    @table_size = table_stats['table_size'].to_i
+    @indexes_size = table_stats['indexes_size'].to_i
+    @total_size = table_stats['total_size'].to_i
+  end
+
+  def self.stats(weeks: 5)
+    # This query retrieve all the table where the column created_at is present
+    created_at_tables_sql = %{
+      SELECT
+        c.relname as name
+      FROM
+        pg_class AS c
+      INNER JOIN
+        pg_attribute AS a on a.attrelid = c.oid
+      WHERE
+        a.attname = 'created_at'
+        AND c.relkind = 'r'
+    }
+
+    starting_date = weeks.weeks.ago.at_beginning_of_week
+
+    # We want to prepare the SQL queries to count the rows in the retrieved tables
+    rows_count_queries = {}
+    ActiveRecord::Base.connection.execute(created_at_tables_sql).to_a.each do |row|
+      # We do an exception for test results table to make some performance improvement
+      rows_count_queries[row['name']] = if row['name'] == 'test_results'
+        Organization
+          .select("
+            SUM(test_payloads.results_count) AS count,
+            date_trunc('week', test_payloads.processed_at) AS trend_date
+          ")
+          .joins('
+            LEFT OUTER JOIN "projects" ON "projects"."organization_id" = "organizations"."id"
+            LEFT OUTER JOIN "project_versions" ON "project_versions"."project_id" = "projects"."id"
+            LEFT OUTER JOIN"test_payloads" ON "test_payloads"."project_version_id" = "project_versions"."id"
+          ')
+          .group("date_trunc('week', test_payloads.processed_at)")
+          .order('trend_date ASC')
+          .where("date_trunc('week', test_payloads.processed_at) >= ?", starting_date)
+      else
+        # We prepare the SQL statement
+        %{
+          SELECT
+            COUNT(created_at) AS count,
+            date_trunc('week', created_at) AS trend_date
+          FROM
+            #{row['name']}
+          WHERE
+            created_at >= $1
+          GROUP BY
+            date_trunc('week', created_at)
+          ORDER BY
+            trend_date ASC
+        }
+      end
+    end
+
+    # Global query to get approximations of the database metrics (rows and disk space)
+    db_stats_sql = %{
+      SELECT
+        short_table_name AS name,
+        pg_table_size(table_name) AS table_size,
+        pg_indexes_size(table_name) AS indexes_size,
+        pg_total_relation_size(table_name) AS total_size,
+        (SELECT reltuples::BIGINT FROM pg_class WHERE relname=short_table_name) AS count
+      FROM (
+        SELECT
+          table_name AS short_table_name,
+          ('"' || table_schema || '"."' || table_name || '"') AS table_name
+        FROM
+          information_schema.tables
+        WHERE
+          table_schema LIKE 'public'
+      ) AS all_tables
+      ORDER BY total_size DESC
```

Comment:
Add a second order criteria on the name to have predictable ordering of tables with the same total size.